src/Transports.AspNetCore.NewtonsoftJson/InternalGraphQLRequest.cs

```diff
@@ -0,0 +1,21 @@
+using GraphQL.Server.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace GraphQL.Server.Transports.AspNetCore.NewtonsoftJson
+{
+    /// <summary>
+    /// A type for deserializing directly into that suits the NewtonsoftJson serializer.
+    /// </summary>
+    internal class InternalGraphQLRequest
```

Comment:
Good idea to separate request to serialization specific and common one. Although this creates a temporary object.

Reply:
To avoid creating a temporary object, I suppose I could explore these classes inheriting the shared one and using attributes to ignore some things. They'd deserialize into their type, then need to manually set the `Inputs` property with the value. What do you think?

Reply:
I think that we have already gone through this and I like the current version more. Details are hidden inside serializer packages. Outside there is the usual DTO.

Reply:
Allocating one temporary object per request is nothing compared to other things during graphql request pipeline.